lib/containers/basetest.pm

```diff
@@ -0,0 +1,43 @@
+# SUSE's openQA tests
+#
+# Copyright 2021 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Base class for container tests
+# Maintainer: qac team <qa-c@suse.de>
+
+package containers::basetest;
+use base 'opensusebasetest';
+use testapi;
+use strict;
+use warnings;
+use containers::common qw(install_docker_when_needed install_podman_when_needed);
+use version_utils 'get_os_release';
+
+sub containers_factory {
+    my ($self, $runtime) = @_;
+    my $engine;
+
+    die("Egnine already initialized") if ($self->{engine});
```

Comment:
the long-plan is to be able to run a single module with different engine even when it is scheduled twice. this doesnt scale  in this case.

Reply:
It does, and I think it's better to handle that in the test module, not in the factory class.

Reply:
i am not sure if i follow. Even if the `$self->{engine}` is defined the invocation should return itself, right? why to die?

Reply:
see https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/publiccloud/basetest.pm#L26